ml/base/estimator.py

```diff
@@ -52,31 +41,34 @@ def _gen_estimator(self):
         )
         return estimator
 
-    def _gen_model_fn(self):
+    def _gen_model_fn(self, gpu_id=2):
 
         def model_fn(features, labels, mode, params):
```

Comment:
Why is a `model_fn` defined in the prototype class? I assume that the abstract classes are not supposed to have specific implementations.

Reply:
My plan is to let custom components implement the used methods in _gen_model_fn.